src/openforms/payments/views.py

```diff
@@ -235,9 +235,7 @@ def _handle_return(self, request, uuid: str):
                 )
                 raise ParseError(detail="redirect not allowed")
 
-        transaction.on_commit(
-            lambda: update_submission_payment_status.delay(payment.submission.id)
-        )
+        transaction.on_commit(lambda: on_post_submission_event(payment.submission.pk))
```

Comment:
Small (and not that important remark): it is recommended to use `partial` over lambdas to avoid late binding. But I think here this is fine (`payment.submission.pk` isn't going to change)